### PR TITLE
Shadow example

### PR DESCRIPTION
**Pull request by kvark**
~~I believe all the nuts and bolts are in place, just figuring out some details of making the shadow match between the rendering and sampling (hence the WIP).~~

Most importantly, this PR includes crucial fixes and improvements of our resource tracking, plus a few fixes in the bind group and texture creation code.

### CODE REVIEW

**Comment by kvark** [comment]
Nailed the issue: our Vulkan-targeting GLSL should be outputting the depth in range 0 to 1, unlike what cgmath-rs projections are currently producing. See https://github.com/rustgd/cgmath/issues/449

wgpu-shadow

File: wgpu-native/src/device.rs [hunk 1]
```diff
@@ -1581,3 +1621,8 @@ pub extern "C" fn wgpu_buffer_set_sub_data(
 
     device.com_allocator.after_submit(comb);
 }
+
+#[no_mangle]
+pub extern "C" fn wgpu_device_destroy(device_id: BufferId) {
+    HUB.devices.unregister(device_id);
+}
```
Comment (grovesNL) [inline wgpu-native/src/device.rs]: nit: newline

File: gfx-examples/src/framework.rs [hunk 2]
```diff
@@ -123,5 +123,6 @@ pub fn run<E: Example>(title: &str) {
 
         let frame = swap_chain.get_next_texture();
         example.render(&frame, &mut device);
+        running &= !cfg!(feature = "metal-auto-capture");
```
Comment (grovesNL) [inline gfx-examples/src/framework.rs]: I assume the intent is to auto-capture one frame only?

**Comment by kvark** [comment]
Thanks!
The auto-capture thing was needed as a workaround for debugging while our presentation was broken. Now, one frame is captured automatically even without it, but the option is useful for compute.

**Pull request merged**